lib/rules/no-unused-vars.js

```diff
@@ -327,7 +327,10 @@ module.exports = {
             }
 
             if (parent.type === "AssignmentExpression" &&
-                grandparent.type === "ExpressionStatement" &&
+                (
+                    grandparent.type === "ExpressionStatement" ||
+                    grandparent.type === "SequenceExpression"
+                ) &&
```

Comment:
I think we have a false positive here - 

```
let x = 0; foo = (0, x = x+ 1); // false positive
```

This should be valid as other cases here - 

https://github.com/eslint/eslint/blob/1f5d0889264c60dddb6fb07a3b1e43f840e84d57/tests/lib/rules/no-unused-vars.js#L180-L182

Reply:
Fixed.